### ISSUE TITLE
Date format is not respected

### ISSUE THREAD

**Issue by szczukot** [home-assistant/frontend#22900]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

I choose date format:
![image](https://github.com/user-attachments/assets/8994fbcc-e7d3-4f10-b57b-14a2e5e68cdf)

but I see everywhere date notation in the United States format

![image](https://github.com/user-attachments/assets/81bccf57-22cb-4de4-b9cc-1378f8604659)

![image](https://github.com/user-attachments/assets/efdd6bfb-f511-40dd-aeb0-92c329bead6b)

![image](https://github.com/user-attachments/assets/3649a2b0-0e97-4e7a-8ff2-b7dccd37a533)

### Describe the behavior you expected

Displaying the date in the selected format

### Steps to reproduce the issue

everywhere

### What version of Home Assistant Core has the issue?

2024.11.2

### In which browser are you experiencing the issue with?

Google Chrome Version 131.0.6778.70 (Official Build) (64-bit)

### Which operating system are you using to run this browser?

Windows 11

**Comment by karwosts** [comment]
Seems that the date format is _only_ used in places with fully numeric dates, like 12/31/24. 

Dates which are formatted in longer style formats are not affected by date format option.

**Comment by szczukot** [comment]
Changing the language does not change this format either (tested English GB and Polish)

**Comment by Stolowka** [comment]
still relevant

**Comment by RobSmyth** [comment]
To expand on the reported issue (let me know if I ought to raise a separate issue), HASS has location and language settings but does not seem to have any locale setting (e.g: en-AU, es-ES, or es-CL) and selecting "English GB" the dev tools | statistics | fix issue dialog shows gives US date format:

![Image](https://github.com/user-attachments/assets/3dc8d448-f37c-43ea-bf4a-06679f897813)

Here is is using the US date format (mm/dd/yyyy). In the end I found that I must set my user date format explicitly.

Improvements:

- A locale configuration (best options)
- Make system default the much more common dd/mm/yyyy format instead of the US format
- Add system level default date format
- Fix the default format for en-GB to be dd/mm/yyyy

**Comment by lukaas33** [comment]
Hi, we are a group of computer science students that are planning to fix this issue as part of a course. 
Perhaps this issue can be assigned to me?

**Comment by bramkragten** [comment]
> Changing the language does not change this format either (tested English GB and Polish)

I don't think that is true, using English GB + the setting `use language settings` for date format you would get d-m-y.

**Comment by RobSmyth** [comment]
> [@RobSmyth](https://github.com/RobSmyth) it is just using d-m-y when selecting en-gb here, can you check if your issue still exists?
> 
> ![Image](https://github.com/user-attachments/assets/0056d10d-d073-46fb-b30c-5cf250d618bd)

Hi @bramkragten ,

Yes, using en-gb now does give GB date format.

Thank you

**Comment by szczukot** [comment]
still relevant

**Comment by sdgsgdf5** [comment]
Local date format is still not respected.

**Comment by kenguest** [comment]
This is still relevant when adding an event to a calendar. The event dialog box displays the date as mm/dd/yyyy irrespective of Country being set to Ireland under the General settings.

**Comment by Stolowka** [comment]
still active

**Comment by ildar170975** [comment]
As was already stated earlier, the date format settings is not used when a date is shown along with a time - which seems to be very strange since the "Date format" setting is considered as a global setting for everything.

<img width="488" height="156" alt="Image" src="https://github.com/user-attachments/assets/f2298509-c0f2-4429-a823-bcf5c37ed0f5" />

**Comment by ellegamee** [comment]
I'm currently working on a fix for this. I am in the process of reproducing the formatting issues locally to ensure that the user's profile date format settings are correctly respected across the UI.

I will update this thread once I have verified the fix in my local development environment and am ready to submit a PR.

**Comment by ellegamee** [comment]
I have a fix ready for this that refactors the date formatting to correctly respect user profile settings (DMY, MDY, YMD) using date-fns. It also fixes a bug where case-sensitivity caused the settings to always fallback to MDY.

I've tested it locally, but I would love if someone could verify it in their environment as well. You can check out my branch from my fork here: https://github.com/home-assistant/frontend/compare/dev...ellegamee:frontend:fix-date-format-settings-22900

I will open a formal PR shortly with a full technical breakdown.

**Comment by ellegamee** [comment]
I have now opened a Draft PR with a fix and a refactor of the date formatting logic: https://github.com/home-assistant/frontend/pull/51526

Please feel free to review the changes and let me know if you have any questions, suggestions, or if you'd like me to test specific scenarios. I want to make sure this covers all edge cases across different locales.

As soon as I get some initial feedback or a 'thumbs up' that it works in other environments, I will mark it as ready for final review!

**Comment by Stolowka** [comment]
still active
